src/goto-programs/goto_trace.cpp

```diff
@@ -348,6 +348,139 @@ bool is_index_member_symbol(const exprt &src)
     return false;
 }
 
+/// \brief show a compact variant of the goto trace on the console
+/// \param out the output stream
+/// \param ns the namespace
+/// \param goto_trace the trace to be shown
+/// \param options any options, e.g., numerical representation
+void show_compact_goto_trace(
+  messaget::mstreamt &out,
+  const namespacet &ns,
+  const goto_tracet &goto_trace,
+  const trace_optionst &options)
+{
+  std::size_t function_depth = 0;
+
+  for(const auto &step : goto_trace.steps)
+  {
+    if(step.is_function_call())
+      function_depth++;
+    else if(step.is_function_return())
+      function_depth--;
+
+    // hide the hidden ones
+    if(step.hidden)
+      continue;
+
+    switch(step.type)
+    {
+    case goto_trace_stept::typet::ASSERT:
+      if(!step.cond_value)
+      {
+        out << '\n';
+        out << messaget::red << "Violated property:" << messaget::reset << '\n';
+        if(!step.pc->source_location.is_nil())
+        {
+          out << "  " << state_location(step, ns) << '\n';
+        }
```

Comment:
No braces needed (usually a nit pick, but here it's inconsistent with a few lines further down).